src/components/enterprise-subsidy-requests/data/tests/hooks.test.js

```diff
@@ -0,0 +1,164 @@
+      subsidyType: SUBSIDY_TYPE.COUPON,
+    });
+  });
+
+  it('sets subsidyRequestConfiguration to null if customer configuration does not exist', async () => {
+    const error = new Error('Something went wrong.');
+    error.customAttributes = {
+      httpErrorStatus: 404,
+    };
+    service.fetchSubsidyRequestConfiguration.mockRejectedValue(error);
+    const { result, waitForNextUpdate } = renderHook(() => useSubsidyRequestConfiguration('uuid'));
+    await waitForNextUpdate();
+    expect(result.current.subsidyRequestConfiguration).toEqual(null);
+  });
+
+  it('handles any errors', async () => {
+    const error = new Error('Something went wrong.');
+    service.fetchSubsidyRequestConfiguration.mockRejectedValue(error);
+    const { result, waitForNextUpdate } = renderHook(() => useSubsidyRequestConfiguration('uuid'));
+    await waitForNextUpdate();
+    expect(result.current.subsidyRequestConfiguration).toEqual(undefined);
+    expect(logger.logError).toHaveBeenCalledWith(error);
+  });
+});
+
+describe('useSubsidyRequests', () => {
+  afterEach(() => jest.clearAllMocks());
+
+  it.each([null, {
+    subsidyRequestsEnabled: false,
+  }, {
+    subsidyRequestsEnabled: true,
+    subsidyType: undefined,
+  }])('should not do anything if subsidy requests are enabled', async (subsidyRequestsConfiguration) => {
+    renderHook(() => useSubsidyRequests(subsidyRequestsConfiguration));
+    expect(service.fetchCouponCodeRequests).not.toHaveBeenCalled();
+    expect(service.fetchLicenseRequests).not.toHaveBeenCalled();
+  });
+
+  it('should fetch coupon code requests', async () => {
+    service.fetchCouponCodeRequests.mockResolvedValue({
+      data: {
+        results: [
+          {
+            lms_user_id: 1,
+            enterprise_customer_uuid: 'uuid',
+          },
+        ],
+      },
+    });
+
+    const { result, waitForNextUpdate } = renderHook(() => useSubsidyRequests({
+      subsidyRequestsEnabled: true,
+      subsidyType: SUBSIDY_TYPE.COUPON,
+      enterpriseCustomerUuid: 'uuid',
+    }));
+    await waitForNextUpdate({
+      timeout: 3000,
+    });
+    expect(service.fetchCouponCodeRequests).toHaveBeenCalledWith('uuid');
+    expect(service.fetchLicenseRequests).not.toHaveBeenCalled();
+
+    expect(result.current.couponCodeRequests).toEqual(
+      [
+        {
+          lmsUserId: 1,
+          enterpriseCustomerUuid: 'uuid',
+        },
+      ],
+    );
+  });
+
+  it('should fetch coupon code requests', async () => {
+    service.fetchCouponCodeRequests.mockResolvedValue({
+      data: {
+        results: [
+          {
+            lms_user_id: 1,
+            enterprise_customer_uuid: 'uuid',
+          },
+        ],
+      },
+    });
+
+    const { result, waitForNextUpdate } = renderHook(() => useSubsidyRequests({
+      subsidyRequestsEnabled: true,
+      subsidyType: SUBSIDY_TYPE.COUPON,
+      enterpriseCustomerUuid: 'uuid',
+    }));
+    await waitForNextUpdate({
+      timeout: 3000,
```

Comment:
[curious] Does this `timeout` make this one test take 3+ seconds in CI? Curious if it the timeout could be any smaller? Same question for other timeouts like this through the PR.

Reply:
remnant + copypasta